token/program-2022/src/extension.rs

```diff
@@ -16,19 +260,25 @@ pub enum AccountType {
     /// Token holding account with additional extensions
     Account,
 }
-
 impl Default for AccountType {
     fn default() -> Self {
         Self::Uninitialized
     }
 }
+// TODO This kind of stinks, but there's no way to automatically implement Pod
+// on an enum, which almost defeats the purpose of in-place serde!
+// Happy to try something else here though.
+#[allow(unsafe_code)]
+unsafe impl Zeroable for AccountType {}
```

Comment:
Using `num-enum` is my pref here with `try_from` mapping to `AccountType:: Uninitialized` on error to avoid the unsafe